src/java/org/apache/cassandra/index/sai/view/IndexViewManager.java

```diff
@@ -61,7 +61,9 @@ public IndexViewManager(IndexContext context)
     IndexViewManager(IndexContext context, Collection<SSTableIndex> indices)
     {
         this.context = context;
-        this.viewRef.set(new View(context, indices));
+        // A view starts our as non-queryable because C* subsequently creates this IndexViewManager before completing
+        // the index build. Once the build is done, it replaces the view with a queryable one.
+        this.viewRef.set(new View(context, indices, false));
```

Comment:
I wonder if it works on CNDB which skips index build on writer.

On an empty and creates an index, existing CNDB directly marks the index as queryable. But with this change, will read request fail?

Reply:
Thanks for this comment. I was wondering if there was different logic in CNDB. I would tend to think this logic is captured in a test though, right? Here is the test pr https://github.com/riptano/cndb/pull/15531. I haven't reviewed the test failures yet, but I will now. None of them appear directly related to SAI at first glance, though.